### ISSUE TITLE
Tags not working for notes in sub-folders between windows and Ubuntu Linux

### ISSUE THREAD

**Issue by Amourspirit** [pbek/QOwnNotes#358]
I just installed QOwnNotes on Windows 8, Windows 10 and Ubuntu 16.04 LTS.

I created a Projects folder and then I created a sub folder named Test in the Projects Folder.

While using Ubuntu I created notes in the Test folder and added a few tags. All seems to work well until I view the note using QOwnNotes on Windows 8 or 10. The tags are missing.

When I add the tags in Windows they are missing in Ubuntu. The tags are displayed in the Tags Panel but are not displayed on the note.

In short when tags are placed on a note in a sub-folder in Ubuntu they can only be seen in Ubuntu. When tags are placed on a note in a sub-folder in Windows they can only be seen in windows.

I was able to track the problem down using a sqlite browser.

The Issue seems to be in the path separator used on Ubuntu ( **/** ) and Windows ( **\** ).
Checking the **noteTagLink** table I can see that tag info is stored in **note_subfolder_path**.
In Windows the field might contain Projects\Test but in Ubuntu in might be Projects/Test

By editing notes.sqlite database for the notes I was able to get tags to display in Ubuntu when I changed **\** ( _backslash_ ) to **/** ( _forward slash_ ) .
Projects\Test became Projects/Test in the  **note_subfolder_path** field and vise versa for Windows.

**Comment by pbek** [comment]
Thank you for reporting. I thought I had taken care of that, I will take a look at it.

**Comment by pbek** [comment]
Could you please post the debug dump from your debug settings, I need to take a look at your paths.

**Comment by pbek** [comment]
And afterwards could you please go to the note folder settings and set your notes path again.
Does it work now?
Could you then please post an other debug dump?

**Comment by pbek** [comment]
For my last two comments I was both times referring to your Windows installation of QOwnNotes.

**Comment by pbek** [comment]
## 16.10.4
- attempted to fix tagging portability problems between Windows and
  other operating systems

**Comment by pbek** [comment]
There now is a new release, could you please test it and report if the new features work for you?
(you can ignore my other comments if it works now)

**Comment by Amourspirit** [comment]
here is my debug dump from Ubuntu
___
QOwnNotes Debug Information
===========================

## General Info

**Current Date**: `Thu Oct 13 11:42:22 2016`
**Version**: `16.10.3`
**Build date**: `Oct 11 2016`
**Build number**: `249`
**Platform**: `linux`
**Operating System**: `Ubuntu 16.04.1 LTS`
**Build architecture**: `x86_64`
**Current architecture**: `x86_64`
**Release**: `Launchpad PPA`
**Qt Version (build)**: `5.5.1`
**Qt Version (runtime)**: `5.5.1`
**Portable mode**: `no`
**Qt Debug**: `0`
**Locale (system)**: `en_CA`
**Locale (interface)**: `en`
**Icon theme**: `breeze-dark-qownnotes`
**Notes in current note folder**: `8`
**Enabled scripts**: `0`

## Server Info

**serverUrl**: *empty*
**appIsValid**: `no`
**connectionErrorMessage**: *empty*

## Note folders

**currentNoteFolderId**: `2`

### Note folder `Root`

**id**: `1`
**isCurrent**: `no`
**activeTagId**: `0`
**localPath**: `/home/paul/Nextcloud/Notes`
**remotePath**: `Notes`
**isShowSubfolders**: `no`
**activeNoteSubFolder name**: *empty*

### Note folder `General`

**id**: `2`
**isCurrent**: `yes`
**activeTagId**: `0`
**localPath**: `/home/paul/Nextcloud/Notes/General`
**remotePath**: `Notes/General`
**isShowSubfolders**: `yes`
**activeNoteSubFolder name**: *empty*

### Note folder `Programming`

**id**: `3`
**isCurrent**: `no`
**activeTagId**: `-1`
**localPath**: `/home/paul/Nextcloud/Notes/Dev`
**remotePath**: `Notes/Dev`
**isShowSubfolders**: `yes`
**activeNoteSubFolder name**: *empty*

## Settings

**Editor/CurrentSchemaKey**: `EditorColorSchema-cdbf28fc-1ddc-4d13-bb21-6a4043316a2f`
**LastUpdateCheck**: `2016-10-13T11:42:11`
**LinkDialog/geometry**: `<binary data>`
**LogDialog/showAtStartup**: `false`
**MainWindow/geometry**: `<binary data>`
**MainWindow/mainToolBar.iconSize**: `24`
**MainWindow/menuBarGeometry**: `<binary data>`
**MainWindow/noteSubFolderTreeWidgetExpandState-3**: `Projects`
**MainWindow/noteTextEdit.code.font**: `monospace,11,-1,2,50,0,0,0,0,0`
**MainWindow/noteTextEdit.font**: `Ubuntu,11,-1,5,50,0,0,0,0,0`
**MainWindow/noteTextView.code.font**: `monospace,11,-1,2,50,0,0,0,0,0`
**MainWindow/noteTextView.font**: `Ubuntu,11,-1,5,50,0,0,0,0,0`
**MainWindow/noteTextView.rtl**: `false`
**NoteDiffDialog/geometry**: `<binary data>`
**PiwikClientId**: `<hidden>`
**SettingsDialog/geometry**: `<binary data>`
**SettingsDialog/mainSplitterState**: `<binary data>`
**ShowSystemTray**: `false`
**SortingModeAlphabetically**: `false`
**WelcomeDialog/geometry**: `<binary data>`
**allowDifferentNoteFileName**: `false`
**allowOnlyOneAppInstance**: `true`
**appMetrics/disableAppHeartbeat**: `false`
**appMetrics/disableTracking**: `false`
**appMetrics/notificationShown**: `true`
**cryptoKey**: `<hidden>`
**currentNoteFolderId**: `2`
**currentWorkspace**: `086af0ad-c3cb-44b0-857b-d86bb463da41`
**customNoteFileExtensionList**: *empty*
**darkMode**: `true`
**darkModeColors**: `true`
**darkModeTrayIcon**: `true`
**defaultNoteFileExtension**: `md`
**disableAutomaticUpdateDialog**: `true`
**dockWasInitializedOnce**: `true`
**externalEditorPath**: `/opt/sublime_text/sublime_text`
**ignoreAllExternalModifications**: `false`
**ignoreAllExternalNoteFolderChanges**: `false`
**interfaceLanguage**: `en`
**internalIconTheme**: `false`
**itemHeight**: `19`
**markdownHighlightingInterval**: `200`
**networking/ignoreSSLErrors**: `true`
**networking/proxyType**: `2`
**noteSaveIntervalTime**: `10`
**notesPath**: `/home/paul/Nextcloud/Notes/General`
**notifyAllExternalModifications**: `true`
**ownCloud/password**: `<hidden>`
**ownCloud/serverUrl**: *empty*
**ownCloud/todoCalendarBackend**: `0`
**ownCloud/todoCalendarCalDAVPassword**: `<hidden>`
**ownCloud/todoCalendarCalDAVServerUrl**: *empty*
**ownCloud/todoCalendarCalDAVUsername**: *empty*
**ownCloud/todoCalendarEnabledList**: *empty*
**ownCloud/todoCalendarEnabledUrlList**: *empty*
**ownCloud/todoCalendarUrlList**: *empty*
**ownCloud/userName**: *empty*
**previousWorkspace**: *empty*
**recentNoteFolders**: `/home/paul/Nextcloud/Notes/Dev, /home/paul/Nextcloud/Notes`
**savedSearches/noteFolder-2**: `Note 2016-10-12T23.47.38, Note 2016-10-12T23.23.54, Note 2016-10-12T23.21.49, Note 2016-10-12T23.14.46`
**showMenuBar**: `true`
**showStatusBar**: `true`
**workspace-086af0ad-c3cb-44b0-857b-d86bb463da41/name**: `default`
**workspace-086af0ad-c3cb-44b0-857b-d86bb463da41/windowState**: `<binary data>`
**workspaces**: `086af0ad-c3cb-44b0-857b-d86bb463da41`

**Comment by Amourspirit** [comment]
Here is debug dump from one of my windows 8 machines

___

QOwnNotes Debug Information
===========================

## General Info

**Current Date**: `Thu Oct 13 11:45:26 2016`
**Version**: `16.10.3`
**Build date**: `Oct 11 2016`
**Build number**: `2318`
**Platform**: `windows`
**Operating System**: `Windows 8.1`
**Build architecture**: `i386`
**Current architecture**: `x86_64`
**Release**: `AppVeyor`
**Qt Version (build)**: `5.5.1`
**Qt Version (runtime)**: `5.5.1`
**Portable mode**: `no`
**Qt Debug**: `0`
**Locale (system)**: `en_CA`
**Locale (interface)**: `en`
**Icon theme**: `breeze-dark-qownnotes`
**Notes in current note folder**: `237`
**Enabled scripts**: `0`

## Server Info

**serverUrl**: *empty*
**appIsValid**: `no`
**connectionErrorMessage**: *empty*

## Note folders

**currentNoteFolderId**: `2`

### Note folder `General`

**id**: `1`
**isCurrent**: `no`
**activeTagId**: `0`
**localPath**: `D:/Users/Paul/NextCloud/Notes/General`
**remotePath**: `Notes`
**isShowSubfolders**: `yes`
**activeNoteSubFolder name**: *empty*

### Note folder `Programming`

**id**: `2`
**isCurrent**: `yes`
**activeTagId**: `-1`
**localPath**: `D:/Users/Paul/NextCloud/Notes/Dev`
**remotePath**: `Notes/Dev`
**isShowSubfolders**: `yes`
**activeNoteSubFolder name**: `Ahk Snippy`

## Settings

**Editor/CurrentSchemaKey**: `EditorColorSchema-cdbf28fc-1ddc-4d13-bb21-6a4043316a2f`
**LastUpdateCheck**: `2016-10-13T11:44:40`
**LogDialog/showAtStartup**: `false`
**MainWindow/geometry**: `<binary data>`
**MainWindow/mainToolBar.iconSize**: `24`
**MainWindow/menuBarGeometry**: `<binary data>`
**MainWindow/noteSubFolderTreeWidgetExpandState-2**: `Projects`
**MainWindow/noteTextEdit.code.font**: `Courier New,9,-1,2,50,0,0,0,0,0`
**MainWindow/noteTextEdit.font**: `MS Shell Dlg 2,8.25,-1,5,50,0,0,0,0,0`
**MainWindow/noteTextView.code.font**: `Courier New,9,-1,2,50,0,0,0,0,0`
**MainWindow/noteTextView.font**: `MS Shell Dlg 2,8.25,-1,5,50,0,0,0,0,0`
**MainWindow/noteTextView.rtl**: `false`
**PiwikClientId**: `<hidden>`
**SettingsDialog/geometry**: `<binary data>`
**SettingsDialog/mainSplitterState**: `<binary data>`
**ShowSystemTray**: `false`
**SortingModeAlphabetically**: `false`
**UpdateDialog/geometry**: `<binary data>`
**WelcomeDialog/geometry**: `<binary data>`
**allowDifferentNoteFileName**: `false`
**allowOnlyOneAppInstance**: `false`
**appMetrics/disableAppHeartbeat**: `false`
**appMetrics/disableTracking**: `false`
**appMetrics/notificationShown**: `true`
**cryptoKey**: `<hidden>`
**currentNoteFolderId**: `2`
**currentWorkspace**: `0785ebad-c0d3-497a-84a0-bebf1d0140af`
**customNoteFileExtensionList**: *empty*
**darkMode**: `true`
**darkModeColors**: `true`
**darkModeTrayIcon**: `true`
**defaultNoteFileExtension**: `md`
**demoNotesCreated**: `true`
**disableAutomaticUpdateDialog**: `false`
**dockWasInitializedOnce**: `true`
**externalEditorPath**: `C:/Program Files/Sublime Text 3/sublime_text.exe`
**ignoreAllExternalModifications**: `false`
**ignoreAllExternalNoteFolderChanges**: `false`
**interfaceLanguage**: `en`
**internalIconTheme**: `false`
**itemHeight**: `15`
**markdownHighlightingInterval**: `200`
**networking/ignoreSSLErrors**: `true`
**networking/proxyType**: `2`
**noteSaveIntervalTime**: `10`
**notesPath**: `D:/Users/Paul/NextCloud/Notes/Dev`
**notifyAllExternalModifications**: `true`
**ownCloud/password**: `<hidden>`
**ownCloud/serverUrl**: *empty*
**ownCloud/todoCalendarBackend**: `0`
**ownCloud/todoCalendarCalDAVPassword**: `<hidden>`
**ownCloud/todoCalendarCalDAVServerUrl**: *empty*
**ownCloud/todoCalendarCalDAVUsername**: *empty*
**ownCloud/todoCalendarEnabledList**: *empty*
**ownCloud/todoCalendarEnabledUrlList**: *empty*
**ownCloud/todoCalendarUrlList**: *empty*
**ownCloud/userName**: *empty*
**previousWorkspace**: *empty*
**recentNoteFolders**: `D:/Users/Paul/NextCloud/Notes/General`
**showMenuBar**: `true`
**showStatusBar**: `true`
**workspace-0785ebad-c0d3-497a-84a0-bebf1d0140af/name**: `default`
**workspace-0785ebad-c0d3-497a-84a0-bebf1d0140af/windowState**: `<binary data>`
**workspaces**: `0785ebad-c0d3-497a-84a0-bebf1d0140af`

**Comment by Amourspirit** [comment]
I will be upgrading to the new version shortly

**Comment by Amourspirit** [comment]
I have installed the new version. Seems to be working but I am unable to test much right now.

For some reason I am have sync issues with my NextCloud and it results in notes not syncing their full content. ( only the first sync when the note is created is getting synced to other clients, notes.sqlite sometimes does not get synced when updated ).

I have confirmed this is not an issue with QOwnNotes and I am currently working on getting it straightened out.

I will post an update if I can get around these and other current issues.
